src/SFML/Audio/SoundStream.cpp

```diff
@@ -91,6 +92,34 @@ struct SoundStream::Impl
             return;
         }
 
+        // Initialize the custom effect node
+        effectNodeVTable.onProcess =
+            [](ma_node* node, const float** framesIn, ma_uint32* frameCountIn, float** framesOut, ma_uint32* frameCountOut)
+        { static_cast<EffectNode*>(node)->impl->processEffect(framesIn, *frameCountIn, framesOut, *frameCountOut); };
+        effectNodeVTable.onGetRequiredInputFrameCount = nullptr;
+        effectNodeVTable.inputBusCount                = 1;
+        effectNodeVTable.outputBusCount               = 1;
+        effectNodeVTable.flags = MA_NODE_FLAG_CONTINUOUS_PROCESSING | MA_NODE_FLAG_ALLOW_NULL_INPUT;
+
+        const auto     nodeChannelCount = ma_engine_get_channels(engine);
+        ma_node_config nodeConfig       = ma_node_config_init();
+        nodeConfig.vtable               = &effectNodeVTable;
+        nodeConfig.pInputChannels       = &nodeChannelCount;
+        nodeConfig.pOutputChannels      = &nodeChannelCount;
+
+        if (const ma_result result = ma_node_init(ma_engine_get_node_graph(engine), &nodeConfig, nullptr, &effectNode);
+            result != MA_SUCCESS)
+        {
+            err() << "Failed to initialize effect node: " << ma_result_description(result) << std::endl;
+            return;
+        }
+
+        effectNode.impl         = this;
+        effectNode.channelCount = nodeChannelCount;
+
+        // Route the sound through the effect node depending on whether an effect processor is set
+        connectEffect(!!effectProcessor);
```

Comment:
```suggestion
        connectEffect(static_cast<bool>(effectProcessor));
```

Personal preference, a bit longer but easier to understand if not familiar with `!!` idiom